### ISSUE TITLE
Best practices with childs

### ISSUE THREAD

**Issue by psypersky** [pinojs/pino#632]
Im trying to decide the best way to use pino in my app, but Im concerned about the cost of creating too many childs

Option 1) Create one child as a way to store information related to this http call, like `correlationId`, `clientIp`, etc, and send the logger to other services as a parameter the function, the second service could create another child to store more information and send it to a third service

In other words Im using a the creation of childs as a tool for injecting and maintaining context of the current http call

Option 2) Use it like you would use `debug` and create a child at the moment of the Node.js module creation

If this is confusing I can provide examples

**Comment by jsumners** [comment]
Take a look at the benchmarks. The cost of child loggers is negligible -- https://github.com/pinojs/pino/blob/0b83ac19b14db96667ee19acd7d51ffa3b7738fa/docs/benchmarks.md

**Comment by psypersky** [comment]
Thank you @jsumners I had that idea from the benchmarks, just needed a bit of confirmation before start refactoring the whole thingy, 🙏🏽🙏🏽🙏🏽

**Comment by davidmarkclements** [comment]
sweet :+1:
